hbase-client/src/main/java/org/apache/hadoop/hbase/security/provider/BuiltInSaslAuthenticationProvider.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.security.provider;
+
+import org.apache.hadoop.hbase.HBaseInterfaceAudience;
+import org.apache.hadoop.io.Text;
+import org.apache.yetus.audience.InterfaceAudience;
+import org.apache.yetus.audience.InterfaceStability;
+
+/**
+ * Base class for all Apache HBase, built-in {@link SaslAuthenticationProvider}'s to extend.
```

Comment:
What does the 'BuiltIn' mean here? For me, I think this should be IA.Private, as it is the base class for providers inside HBase code. But here you mark it as IA.LimitedPrivate...

Reply:
You interpret "BuiltIn" correctly: it's meant to be the base-class for all of the authentication methods that HBase has out of the box.

Right now, I have all of the implementations marked as `LimitedPrivate` audience. It's probably better to have users copy our code, rather than try to extend it. I like your suggestion (to move them all to `IA.Private`).